unit_1d_hw.py

```diff
@@ -83,30 +83,38 @@ def <function_name>():
 '''
 Create a function called slope, that takes in parameters, m, x and b. Where:
 
-:param m: (int) The slope to scale the the list x
+:param m: (float) The slope to scale the the list x
 :param x: (list) The list of values we want to scale and shift based on m and b
-:param b: (int) The intercept to shift the values of x
+:param b: (float) The intercept to shift the values of x
 
-Have the function scale each value of x then add b and store it back into x.
+Have the function scale each value of x then add b and store the result back into x.
 
 Have the function return x
 
 Hint: Use a for loop to iterate through the list
 '''
 
-##################################
+# Here is some code made for you
 
-def slope(m, x, b):
+def slope(m : float, x : list, b : float):
```

Comment:
I think the proper formatting for this would be:

```
def slope(m: float, x: list[float], b: float) -> list[float]:
```

As for when to introduce it, I think maybe now just starting by exposing the syntax to them and giving a quick overview to them (normalize it for them?) and letting people ask questions about it themselves? The parallels with the list of params above it may help normalize the concept for them.

Reply:
Yes I added this in class. I did not go deeper into the list[float], I honestly forgot how to do the type inside another data structure.